File: backend/tests/Live/LiveGreetingTest.php
<?php

declare(strict_types=1);

namespace App\Tests\Live;

use App\Live\LiveGreeting;
use PHPUnit\Framework\TestCase;

final class LiveGreetingTest extends TestCase
{
    public function testSpokenGreetingIncludesOpeningQuestion(): void
    {
        $spoken = LiveGreeting::spoken('Wat is er aan de hand in uw woning?');
        self::assertStringContainsString('huurwoning', $spoken);
        self::assertStringContainsString('Wat is er aan de hand in uw woning?', $spoken);
        self::assertStringContainsString('Greet immediately', LiveGreeting::instructions($spoken));
        self::assertStringContainsString('Do not switch back to Dutch', LiveGreeting::instructions($spoken));
        self::assertStringNotContainsString('Speak Dutch.', LiveGreeting::instructions($spoken));
        self::assertStringContainsString($spoken, LiveGreeting::commentary($spoken));
    }
}

File: android/app/src/test/java/nl/woningtriage/app/voice/LiveGreetingTest.kt
package nl.woningtriage.app.voice

import org.junit.Assert.assertTrue
import org.junit.Test

class LiveGreetingTest {
    @Test
    fun spokenGreetingIncludesRentalHomeAndQuestion() {
        val spoken = LiveGreeting.spoken("Wat is er aan de hand in uw woning?")
        assertTrue(spoken.contains("huurwoning"))
        assertTrue(spoken.contains("Wat is er aan de hand in uw woning?"))
        assertTrue(LiveGreeting.instructions(spoken).contains("Greet immediately"))
        assertTrue(LiveGreeting.instructions(spoken).contains("Do not switch back to Dutch"))
        assertTrue(!LiveGreeting.instructions(spoken).contains("Speak Dutch."))
        assertTrue(LiveGreeting.commentary(spoken).contains(spoken))
    }
}
